File: app/src/main/java/com/example/retube/models/HomeMostPopular.java
package com.example.retube.models;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class HomeMostPopular {

    @SerializedName("items")
    @Expose
    private List<Item> items = null;

    public List<Item> getItems() {
        return items;
    }

    public void setItems(List<Item> items) {
        this.items = items;
    }

    public class Default {

        @SerializedName("url")
        @Expose
        private String url;
        @SerializedName("width")
        @Expose
        private Integer width;
        @SerializedName("height")
        @Expose
        private Integer height;

        public String getUrl() {
            return url;
        }

        public void setUrl(String url) {
            this.url = url;
        }

        public Integer getWidth() {
            return width;
        }

        public void setWidth(Integer width) {
            this.width = width;
        }

        public Integer getHeight() {
            return height;
        }

        public void setHeight(Integer height) {
            this.height = height;
        }

    }


    public class High {

        @SerializedName("url")
        @Expose
        private String url;
        @SerializedName("width")
        @Expose
        private Integer width;
        @SerializedName("height")
        @Expose
        private Integer height;

        public String getUrl() {
            return url;
        }

        public void setUrl(String url) {
            this.url = url;
        }

        public Integer getWidth() {
            return width;
        }

        public void setWidth(Integer width) {
            this.width = width;
        }

        public Integer getHeight() {
            return height;
        }

        public void setHeight(Integer height) {
            this.height = height;
        }

    }

    public class Item {

        @SerializedName("id")
        @Expose
        private String id;
        @SerializedName("snippet")
        @Expose
        private Snippet snippet;
        @SerializedName("statistics")
        @Expose
        private Statistics statistics;

        public String getId() {
            return id;
        }

        public void setId(String id) {
            this.id = id;
        }

        public Snippet getSnippet() {
            return snippet;
        }

        public void setSnippet(Snippet snippet) {
            this.snippet = snippet;
        }

        public Statistics getStatistics() {
            return statistics;
        }

        public void setStatistics(Statistics statistics) {
            this.statistics = statistics;
        }

    }


    public class Maxres {

        @SerializedName("url")
        @Expose
        private String url;
        @SerializedName("width")
        @Expose
        private Integer width;
        @SerializedName("height")
        @Expose
        private Integer height;

        public String getUrl() {
            return url;
        }

        public void setUrl(String url) {
            this.url = url;
        }

        public Integer getWidth() {
            return width;
        }

        public void setWidth(Integer width) {
            this.width = width;
        }

        public Integer getHeight() {
            return height;
        }

        public void setHeight(Integer height) {
            this.height = height;
        }

    }


    public class Medium {

        @SerializedName("url")
        @Expose
        private String url;
        @SerializedName("width")
        @Expose
        private Integer width;
        @SerializedName("height")
        @Expose
        private Integer height;

        public String getUrl() {
            return url;
        }

        public void setUrl(String url) {
            this.url = url;
        }

        public Integer getWidth() {
            return width;
        }

        public void setWidth(Integer width) {
            this.width = width;
        }

        public Integer getHeight() {
            return height;
        }

        public void setHeight(Integer height) {
            this.height = height;
        }

    }


    public class Snippet {

        @SerializedName("publishedAt")
        @Expose
        private String publishedAt;
        @SerializedName("channelId")
        @Expose
        private String channelId;
        @SerializedName("title")
        @Expose
        private String title;
        @SerializedName("thumbnails")
        @Expose
        private Thumbnails thumbnails;
        @SerializedName("channelTitle")
        @Expose
        private String channelTitle;

        public String getPublishedAt() {
            return publishedAt;
        }

        public void setPublishedAt(String publishedAt) {
            this.publishedAt = publishedAt;
        }

        public String getChannelId() {
            return channelId;
        }

        public void setChannelId(String channelId) {
            this.channelId = channelId;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }

        public Thumbnails getThumbnails() {
            return thumbnails;
        }

        public void setThumbnails(Thumbnails thumbnails) {
            this.thumbnails = thumbnails;
        }

        public String getChannelTitle() {
            return channelTitle;
        }

        public void setChannelTitle(String channelTitle) {
            this.channelTitle = channelTitle;
        }

    }


    public class Standard {

        @SerializedName("url")
        @Expose
        private String url;
        @SerializedName("width")
        @Expose
        private Integer width;
        @SerializedName("height")
        @Expose
        private Integer height;

        public String getUrl() {
            return url;
        }

        public void setUrl(String url) {
            this.url = url;
        }

        public Integer getWidth() {
            return width;
        }

        public void setWidth(Integer width) {
            this.width = width;
        }

        public Integer getHeight() {
            return height;
        }

        public void setHeight(Integer height) {
            this.height = height;
        }

    }


    public class Statistics {

        @SerializedName("viewCount")
        @Expose
        private String viewCount;
        @SerializedName("likeCount")
        @Expose
        private String likeCount;
        @SerializedName("dislikeCount")
        @Expose
        private String dislikeCount;
        @SerializedName("favoriteCount")
        @Expose
        private String favoriteCount;
        @SerializedName("commentCount")
        @Expose
        private String commentCount;

        public String getViewCount() {
            return viewCount;
        }

        public void setViewCount(String viewCount) {
            this.viewCount = viewCount;
        }

        public String getLikeCount() {
            return likeCount;
        }

        public void setLikeCount(String likeCount) {
            this.likeCount = likeCount;
        }

        public String getDislikeCount() {
            return dislikeCount;
        }

        public void setDislikeCount(String dislikeCount) {
            this.dislikeCount = dislikeCount;
        }

        public String getFavoriteCount() {
            return favoriteCount;
        }

        public void setFavoriteCount(String favoriteCount) {
            this.favoriteCount = favoriteCount;
        }

        public String getCommentCount() {
            return commentCount;
        }

        public void setCommentCount(String commentCount) {
            this.commentCount = commentCount;
        }

    }


    public class Thumbnails {

        @SerializedName("default")
        @Expose
        private Default _default;
        @SerializedName("medium")
        @Expose
        private Medium medium;
        @SerializedName("high")
        @Expose
        private High high;
        @SerializedName("standard")
        @Expose
        private Standard standard;
        @SerializedName("maxres")
        @Expose
        private Maxres maxres;

        public Default getDefault() {
            return _default;
        }

        public void setDefault(Default _default) {
            this._default = _default;
        }

        public Medium getMedium() {
            return medium;
        }

        public void setMedium(Medium medium) {
            this.medium = medium;
        }

        public High getHigh() {
            return high;
        }

        public void setHigh(High high) {
            this.high = high;
        }

        public Standard getStandard() {
            return standard;
        }

        public void setStandard(Standard standard) {
            this.standard = standard;
        }

        public Maxres getMaxres() {
            return maxres;
        }

        public void setMaxres(Maxres maxres) {
            this.maxres = maxres;
        }

    }
}

File: app/src/main/java/lej/happy/retube/data/models/youtube/HomeMostPopular.kt
package lej.happy.retube.data.models.youtube

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName



data class HomeMostPopular (

	@SerializedName("items")
	@Expose
	val items : List<Items>
){

	data class Items (

		@SerializedName("id")
		@Expose
		val id : String,

		@SerializedName("snippet")
		@Expose
		val snippet : Snippet,

		@SerializedName("statistics")
		@Expose
		val statistics : Statistics
	)

	data class Snippet (

		@SerializedName("publishedAt")
		@Expose
		val publishedAt : String,

		@SerializedName("channelId")
		@Expose
		val channelId : String,
		@SerializedName("title")
		@Expose
		val title : String,
		@SerializedName("thumbnails")
		@Expose
		val thumbnails : Thumbnails,
		@SerializedName("channelTitle")
		@Expose
		val channelTitle : String
	)

	data class Statistics (

		@SerializedName("viewCount")
		@Expose
		val viewCount : Int,
		@SerializedName("likeCount")
		@Expose
		val likeCount : Int,
		@SerializedName("dislikeCount")
		@Expose
		val dislikeCount : Int,
		@SerializedName("favoriteCount")
		@Expose
		val favoriteCount : Int,
		@SerializedName("commentCount")
		@Expose
		val commentCount : Int
	)


	data class Thumbnails (

		@SerializedName("default")
		@Expose
		val default : Default,
		@SerializedName("medium")
		@Expose
		val medium : Medium,
		@SerializedName("high")
		@Expose
		val high : High,
		@SerializedName("standard")
		@Expose
		val standard : Standard,
		@SerializedName("maxres")
		@Expose
		val maxres : Maxres
	)

	data class High (

		@SerializedName("url")
		@Expose
		val url : String,
		@SerializedName("width")
		@Expose
		val width : Int,
		@SerializedName("height")
		@Expose
		val height : Int
	)

	data class Default (

		@SerializedName("url")
		@Expose
		val url : String,
		@SerializedName("width")
		@Expose
		val width : Int,
		@SerializedName("height")
		@Expose
		val height : Int
	)

	data class Medium (

		@SerializedName("url")
		@Expose
		val url : String,
		@SerializedName("width")
		@Expose
		val width : Int,
		@SerializedName("height")
		@Expose
		val height : Int
	)

	data class Standard (

		@SerializedName("url")
		@Expose
		val url : String,
		@SerializedName("width")
		@Expose
		val width : Int,
		@SerializedName("height")
		@Expose
		val height : Int
	)

	data class Maxres (

		@SerializedName("url")
		@Expose
		val url : String,
		@SerializedName("width")
		@Expose
		val width : Int,
		@SerializedName("height")
		@Expose
		val height : Int
	)

}
